vaccine/vaccine_reg_ussd.py

```diff
@@ -0,0 +1,363 @@
+                assert value.isdigit()
+                assert int(value) > date.today().year - MAX_AGE
+                assert int(value) <= date.today().year
+            except AssertionError:
+                raise ErrorMessage(
+                    "REQUIRED: Please TYPE the 4 digits of the year you were born in "
+                    "(Example: 1980)"
+                )
+
+        return FreeText(
+            self,
+            question="DOB: In what year were you born? (Please type)",
+            next="state_dob_month",
+            check=validate_dob_year,
+        )
+
+    async def state_dob_month(self):
+        return ChoiceState(
+            self,
+            question="DOB: Select your month",
+            choices=[
+                Choice("1", "Jan"),
+                Choice("2", "Feb"),
+                Choice("3", "Mar"),
+                Choice("4", "Apr"),
+                Choice("5", "May"),
+                Choice("6", "June"),
+                Choice("7", "July"),
+                Choice("8", "Aug"),
+                Choice("9", "Sep"),
+                Choice("10", "Oct"),
+                Choice("11", "Nov"),
+                Choice("12", "Dec"),
+            ],
+            next="state_dob_day",
+            error="REQUIRED: In which month were you born?",
+        )
+
+    async def state_dob_day(self):
+        # TODO: stop <40 year olds from continuing
+        async def validate_dob_day(value):
+            dob_year = int(self.user.answers["state_dob_year"])
+            dob_month = int(self.user.answers["state_dob_month"])
+            try:
+                assert value.isdigit()
+                date(dob_year, dob_month, int(value))
+            except (AssertionError, ValueError):
+                raise ErrorMessage(
+                    "ERROR: Please reply with DAY of your birthday Example: 20"
+                )
+
+        return FreeText(
+            self,
+            question="DOB: Which day of the month were you born on",
+            next="state_first_name",
+            check=validate_dob_day,
+        )
+
+    async def state_first_name(self):
+        return FreeText(
+            self, question="Please enter your FIRST name", next="state_surname"
+        )
+
+    async def state_surname(self):
+        return FreeText(
+            self, question="Please enter your SURNAME", next="state_confirm_profile"
+        )
+
+    async def state_confirm_profile(self):
+        first_name = self.user.answers["state_first_name"][:27]
+        surname = self.user.answers["state_surname"][:27]
+        id_type = self.ID_TYPES[self.user.answers["state_identification_type"]].value
+        id_number = self.user.answers["state_identification_number"][:20]
+        return MenuState(
+            self,
+            question="\n".join(
+                [
+                    "Confirm the following:",
+                    "",
+                    f"{first_name} {surname}",
+                    id_type,
```

Comment:
We were not going to include the id_type in the confirmation question. Also, how many characters do we have to play around in this screen? Makes sense to remove this and only confirm Firstname, Surname, ID

Reply:
So I've limited first name and last name to 27 characters, and identification to 20 characters. For the longest id type, on the error message (that's longer than the question message), that gives us 158 characters if all those fields are at their maximum, to 2 characters to spare. But we can adjust those maximums.

Reply:
Ok, but I think we should remove `id_type` and increase the limits accordingly.

Reply:
@Neville-Tietz Please review this suggestion - it has copy implications

Reply:
@miltontony @rudigiesler Can we drop id_type but include a shortened type as in "ID#:identification_number"